File: 操作系统/Windows API/Windows 程序设计/第6版/CPlusPlus/Chapter08/SegoeSymbols/SegoeSymbols/DoubleToStringHexByteConverter.cpp
#include "pch.h"
#include "DoubleToStringHexByteConverter.h"

using namespace SegoeSymbols;

using namespace Platform;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml;
using namespace Windows::UI::Xaml::Interop;

Object^ DoubleToStringHexByteConverter::Convert(Object^ value, TypeName targetType, Object^ parameter, String^)
{
    wchar_t result[6];
    swprintf(result, sizeof(result) / sizeof(result[0]), L"%02X__", (int)(double)value);
    return ref new String(result);
}

Object^ DoubleToStringHexByteConverter::ConvertBack(Object^ value, TypeName targetType, Object^ parameter, String^)
{
    return value;
}

File: 操作系统/Windows API/Windows 程序设计/第6版/CSharp/Chapter08/SegoeSymbols/SegoeSymbols/DoubleToStringHexByteConverter.cs
using System;
using Windows.UI.Xaml.Data;

namespace SegoeSymbols
{
    public class DoubleToStringHexByteConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return ((int)(double)value).ToString("X2") + "__";
        }
        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return value;
        }
    }
}
